### PR TITLE
feat: inline rules with extra actions

### LINKED ISSUE #267: Allow to add extra actions on inline rules
**Is your feature request related to a problem? Please describe.**
Right now inline rules are allowed with the following syntax:

```yml
workflows:
  - name: size-small
    if:
      - "$size < 3"   # inline rule
    then:
      - $addLabel("small")
  
  - name: size-medium # inline rule
    if: 
      - "$size >= 3 && $size < 5"
    then:
      - $addLabel("medium")
  
  - name: size-large  # inline rule
    if:
      - "$size > 5"
    then:
      - $addLabel("large")
```

Although this solution works fine we want it to work with `extra-actions`.

**Describe the solution you'd like**
We should be able to also define an inline rule with `rule` and add `extra-actions`.

Therefore the following workflow `size` should be allowed:
```yml
workflows:
  - name: size
    if:
      - rule: "$size < 3"               # inline rule
        extra-actions:
          - $addLabel("small")
      - rule: "$size >= 3 && $size < 5" # inline rule
        extra-actions:
          - $addLabel("medium")
      - rule: "$size > 5"               # inline rule
        extra-actions:
          - $addLabel("large")
```

### PR DESCRIPTION
**Pull request by zolamk**
## Description

This PR introduces support for inline rules with extra actions like below

```
workflows:
  - name: size
    if:
      - rule: "$size < 3"               # inline rule
        extra-actions:
          - $addLabel("small")
      - rule: "$size >= 3 && $size < 5" # inline rule
        extra-actions:
          - $addLabel("medium")
      - rule: "$size > 5"               # inline rule
        extra-actions:
          - $addLabel("large")
 ```
How this is implemented is during the loading stage we check if the rule exists in the Reviewpad file and if not we add it to the list of registered rules and refer the newly registered rule in the workflow

## Related issue

Closes #267

## Type of change

New feature (non-breaking change which adds functionality) 

## How was this tested?
Unit Tests and Manual tests

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: engine/loader.go [hunk 1]
```diff
@@ -242,24 +242,34 @@ func processInlineRulesOnWorkflow(workflow PadWorkflow) (*PadWorkflow, []PadRule
...
+				Rule: padRule.Name,
 			})
 		case map[string]interface{}:
 			rule, err := mapToPadWorkflowRule(r)
 			if err != nil {
 				return nil, nil, err
 			}
 
+			// if the rule doesn't exist in the list of rules
+			// treat it as an inline rule with extra actions
```
Comment (ferreiratiago) [inline engine/loader.go:259]:
```suggestion
			// if the rule doesn't exist in the list of available rules
			// treat it as an inline rule
```

File: engine/loader.go [hunk 2]
```diff
@@ -242,24 +242,34 @@ func processInlineRulesOnWorkflow(workflow PadWorkflow) (*PadWorkflow, []PadRule
...
+			if _, exists := findRule(rules, rule.Rule); !exists {
+				padRule := inlineRuleToPadRule(rule.Rule)
+
+				foundInlineRules = append(foundInlineRules, padRule)
+
+				wf.Rules = append(wf.Rules, PadWorkflowRule{
+					Rule:         padRule.Name,
+					ExtraActions: rule.ExtraActions,
```
Comment (ferreiratiago) [inline engine/loader.go:267]: Having extra action is not mandatory. Please make sure that this works without `extra-actions`.

e.g.
```yml
workflows:
  - name: test-process
    if:
      - rule: '$size() <= 30'
        extra-actions:
          - $addLabel("small")
      - rule: '$size() > 30'
    then:
      - '$addLabel("critical")'
```

**Review by ferreiratiago — CHANGES_REQUESTED** [review]
Didn't look into the tests but please go ahead with the suggested recommendations.

File: engine/loader.go (hunk 2 above)
Comment (zolamk) [reply engine/loader.go:267]: Yeah, I've checked and confirmed it works properly

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**